### PR TITLE
Adds bread smite and transformation animation helper.

### PR DESCRIPTION
**Pull request by AnturK**
Adds bread smite, and generic transformation animation helper for the future.

https://streamable.com/2m7f9o

### CODE REVIEW

**Comment by trollbreeder** [comment]
What?? Corgi is bread??? HOW?!

**Comment by Qustinnus** [comment]
image

File: code/__HELPERS/icons.dm [hunk 1]
```diff
@@ -1220,3 +1220,46 @@ GLOBAL_DATUM_INIT(dummySave, /savefile, new("tmp/dummySave.sav")) //Cache of ico
 
 	var/icon/I = getFlatIcon(thing)
 	return icon2html(I, target)
+
+GLOBAL_LIST_EMPTY(transformation_animation_objects)
+
+/atom/movable/proc/transformation_animation(result_appearance,time = 3 SECONDS,transform_overlay=FALSE,reset_after=TRUE)
```
Comment (Qustinnus) [inline code/__HELPERS/icons.dm:1226]: add some autodoc and explanation on how to use this else it'll end up going into obscurity

File: code/__HELPERS/icons.dm [hunk 2]
```diff
@@ -1220,3 +1220,46 @@ GLOBAL_DATUM_INIT(dummySave, /savefile, new("tmp/dummySave.sav")) //Cache of ico
...
+		animate(transform_effect,pixel_y=-16,time=time)
+		animate(alpha=0)
+
+	GLOB.transformation_animation_objects[src] = transformation_objects
+	for(var/A in transformation_objects)
+		vis_contents += A
+	if(reset_after)
+		addtimer(CALLBACK(src,.proc/reset_transformation_animation),time)
+
+/atom/movable/proc/reset_transformation_animation()
```
Comment (Qustinnus) [inline code/__HELPERS/icons.dm:1258]: ditto

File: code/modules/admin/verbs/randomverbs.dm [hunk 3]
```diff
@@ -1238,9 +1239,17 @@ Traitors and the like can also be revived with the previous role mostly intact.
 				to_chat(usr,"<span class='warning'>[dude] does not have knottable shoes!</span>", confidential = TRUE)
 				return
 			sick_kicks.adjust_laces(SHOES_KNOTTED)
+		if(ADMIN_PUNISHMENT_BREAD)
+			var/mutable_appearance/bread_appearance = mutable_appearance('icons/obj/food/burgerbread.dmi',"bread")
+			target.transformation_animation(bread_appearance,time= 5 SECONDS,transform_overlay=TRUE,reset_after=TRUE)
+			addtimer(CALLBACK(GLOBAL_PROC, .proc/breadify, target), 5 SECONDS)
```
Comment (Qustinnus) [inline code/modules/admin/verbs/randomverbs.dm:1245]: wouldn't adding a callback to the transform animation be better than this?

File: code/__HELPERS/icons.dm [hunk 4]
```diff
@@ -1220,3 +1220,46 @@ GLOBAL_DATUM_INIT(dummySave, /savefile, new("tmp/dummySave.sav")) //Cache of ico
...
+GLOBAL_LIST_EMPTY(transformation_animation_objects)
+
+/atom/movable/proc/transformation_animation(result_appearance,time = 3 SECONDS,transform_overlay=FALSE,reset_after=TRUE)
+	var/list/transformation_objects = GLOB.transformation_animation_objects[src] || list()
+	//Disappearing part
+	var/top_part_filter = filter(type="alpha",icon=icon('icons/effects/alphacolors.dmi',"white"),y=0)
+	filters += top_part_filter
+	animate(filters[length(filters)],y=-32,time=time)
+	//Appearing part
+	var/obj/effect/overlay/ugh = new
```
Comment (Jared-Fogle) [inline code/__HELPERS/icons.dm:1233]: Can this have a better variable name?

**Comment by AnturK** [comment]
WIP till i figure out if 1529 broke it

**Comment by LemonInTheDark** [comment]
Re-ready?

**Comment by AnturK** [comment]
Yeah should be good now.

**Comment by LemonInTheDark** [comment]
I'm scared

**Comment by TheVekter** [comment]
I used this for the first time tonight, it's _amazing_.

**Pull request merged**